src/Synchronizer/VariantConfigurationSynchronizer.php

```diff
@@ -22,27 +24,45 @@ final class VariantConfigurationSynchronizer implements VariantConfigurationSync
 
     private VariantCollectionInterface $variantCollection;
 
+    private VariantGeneratorRegistryInterface $generatorRegistry;
+
     public function __construct(
         FactoryInterface $variantConfigurationFactory,
         VariantConfigurationRepositoryInterface $variantConfigurationRepository,
         VariantCollectionInterface $variantCollection,
-        ManagerRegistry $managerRegistry
+        ManagerRegistry $managerRegistry,
+        VariantGeneratorRegistryInterface $generatorRegistry
     ) {
         $this->variantConfigurationFactory = $variantConfigurationFactory;
         $this->variantConfigurationRepository = $variantConfigurationRepository;
         $this->variantCollection = $variantCollection;
         $this->managerRegistry = $managerRegistry;
+        $this->generatorRegistry = $generatorRegistry;
     }
 
-    public function synchronize(): void
+    public function synchronize(bool $runSetup): VariantConfigurationSynchronizationResultInterface
```

Comment:
This `$runSetup` maybe have a default value? I.e. `bool $runSetup = true`

Reply:
Do you want to have that specified at the interface level or just on this implementation?

Reply:
Yeah, on the interface